docs/api/websocket.rst

```diff
@@ -0,0 +1,505 @@
+            """
+
+        async def process_resource_ws(self, req, ws, resource, params):
+            """Process a WebSocket handshake request after routing.
+
+            Note:
+                This method is only called when the request matches
+                a route to a resource.
+
+            Args:
+                req: Request object that will be passed to the
+                    routed responder.
+                ws: WebSocket object that will be passed to the
+                    routed responder.
+                resource: Resource object to which the request was
+                    routed.
+                params: A dict-like object representing any additional
+                    params derived from the route's URI template fields,
+                    that will be passed to the resource's responder
+                    method as keyword arguments.
+            """
+
+If a route is found for the requested path, the framework will then check for
+a responder coroutine named ``on_websocket()`` on the target resource. If the
+responder is found, it is invoked in a similar manner to a regular
+``on_get()`` responder, except that a :class:`falcon.asgi.WebSocket` object
+is passed in, instead of an object of type :class:`falcon.asgi.Response`.
+
+For example, given a route that includes an ``account_id`` path parameter, the
+framework would expect an ``on_websocket()`` responder similar to this:
+
+.. code:: python
+
+    async def on_websocket(self, req: Request, ws: WebSocket, account_id: str):
+        pass
+
+If no route matches the path requested in the WebSocket handshake, control then
+passes to a default responder that simply raises an instance of
+:class:`~.HTTPRouteNotFound`. By default, this error will be rendered as a 403
+response with a 3404 close code. This behavior can be modified by adding a
+custom error handler (see also: :meth:`~falcon.asgi.App.add_error_handler`).
+
+Similarly, if a route exists but the target resource does not implement an
+``on_websocket()`` responder, the framework invokes a default responder that
+raises an instance of :class:`~.HTTPMethodNotAllowed`. This class will be
+rendered by default as a 403 response with a 3405 close code.
+
+.. _ws_lost_connection:
+
+Lost Connections
+----------------
+
+When the app attempts to receive a message from the client, the ASGI server
+emits a `disconnect` event if the connection has been lost for any reason. Falcon
+surfaces this event by raising an instance of :class:`~.WebSocketDisconnected`
+to the caller.
+
+On the other hand, the ASGI spec requires the ASGI server to silently consume
+messages sent by the app after the connection has been lost (i.e.,  it should
+not be considered an error). Therefore, an endpoint that primarily streams
+outbound events to the client might continue consuming resources unnecessarily
+for some time after the connection is lost.
+
+As a workaround, Falcon implements a small incoming message queue that is used
```

Comment:
I'm not really sold by workaround in this context. Maybe mitigation could be better?

Reply:
IMHO, _workaround_ is fine. _Mitigation_ would suit better if we could mitigate buffer bloat, but we simply can not.

_Workaround_ here means that the ASGI spec does not afford uniform signalling of client disconnect events in all cases, but we work around that issue with an own layer of logic. I would perceive _mitigation_ as more connected to patching security holes, reducing harmful effects.

Reply:
I like "mitigation".